pkg/api/testing/bazel.go

```diff
@@ -42,10 +42,16 @@ func PathForCRD(t *testing.T, name string) string {
 
 func CRDDirectory(t *testing.T) string {
 	runfiles := os.Getenv("RUNFILES_DIR")
-	if runfiles == "" {
-		t.Fatalf("integration tests can only run within 'bazel test' environment")
+	bazelDir := os.Getenv("BAZEL_BIN_DIR")
+	if runfiles == "" && bazelDir == "" {
+		t.Fatalf("integration tests can only run within 'bazel test' environment or have BAZEL_BIN_DIR set")
+	}
+	var path string
+	if bazelDir != "" {
+		path = filepath.Join(bazelDir, "deploy", "crds")
```

Comment:
Is the intention here that someone generates (out of band) their own CRD manifests and places them in a directory under deploy/crds whilst setting their BAZEL_BIN_DIR explicitly?

It doesn’t seem obvious that this is how to configure things to me 👀 could the 'running outside bazel' approach be reduced to having someone run a script which outputs something like "test dependencies now all ready, run: 'export X=...' to prepare your environment" and then the rest just works?

Reply:
Ah I see that is what is happening already! 😀 Maybe just a comment here to explain why and when RUNFILES vs BIN will be used? 🙂

Reply:
Added a note!